src/sql-parser/tests/testdata/ddl

```diff
@@ -1477,6 +1477,13 @@ error: Expected one of SNAPSHOT or VERSION, found SIZE
 ALTER SINK name RESET (SIZE)
                        ^
 
+parse-statement
+ALTER SINK name FROM foobar
```

Comment:
PostgreSQL `ALTER` commands are generally verbs:

  * `ALTER ... SET SCHEMA`
  * `ALTER ... SET TABLESPACE`
  * `ALTER ... ATTACH PERMISSION`
  * `ALTER ... RENAME TO`
  * `ALTER ... OWNER TO` (ok not quite a verb, but they're trying to make `OWNER` one)

`ALTER ... SET SCHEMA` is the most similar prior art. 

I propose we follow suit and use `ALTER ... SET FROM` as the syntax. This is in line with the syntax we had in mind for changing an object's cluster: `ALTER ... SET CLUSTER`.